packages/dd-trace/src/telemetry/index.js

```diff
@@ -1,393 +1,40 @@
-    Object.keys(extendedHeartbeatPayload).forEach(key => delete extendedHeartbeatPayload[key])
-  }, 1000 * 60 * 60 * 24).unref()
-  return extendedInterval
-}
-
-function start (aConfig, thePluginManager) {
-  if (!aConfig.telemetry.enabled) {
-    if (aConfig.sca?.enabled) {
-      logger.warn('DD_APPSEC_SCA_ENABLED requires enabling telemetry to work.')
-    }
-
-    return
-  }
-  config = aConfig
-  pluginManager = thePluginManager
-  application = createAppObject(config)
-  host = createHostObject()
-  heartbeatInterval = config.telemetry.heartbeatInterval
-  integrations = getIntegrations()
-
-  dependencies.start(config, application, host, getRetryData, updateRetryData)
-  telemetryLogger.start(config)
-
-  sendData(config, application, host, 'app-started', appStarted(config))
-
-  if (integrations.length > 0) {
-    sendData(config, application, host, 'app-integrations-change',
-      { integrations }, updateRetryData)
-  }
-
-  heartbeat(config, application, host)
-
-  extendedHeartbeat(config)
-
-  process.on('beforeExit', onBeforeExit)
-  telemetryStartChannel.publish(getTelemetryData())
-}
-
-function stop () {
-  if (!config) {
-    return
-  }
-  clearInterval(extendedInterval)
-  clearTimeout(heartbeatTimeout)
-  process.removeListener('beforeExit', onBeforeExit)
-
-  telemetryStopChannel.publish(getTelemetryData())
-
-  config = undefined
-}
-
-function updateIntegrations () {
-  if (!config || !config.telemetry.enabled) {
-    return
-  }
-  const integrations = getIntegrations()
-  if (integrations.length === 0) {
-    return
-  }
-
-  const { reqType, payload } = createPayload('app-integrations-change', { integrations })
-
-  sendData(config, application, host, reqType, payload, updateRetryData)
-}
-
-function formatMapForTelemetry (map) {
-  // format from an object to a string map in order for
-  // telemetry intake to accept the configuration
-  return map
-    ? Object.entries(map).map(([key, value]) => `${key}:${value}`).join(',')
-    : ''
-}
-
-function updateConfig (changes, config) {
-  if (!config.telemetry.enabled) return
-  if (changes.length === 0) return
-
-  logger.trace(changes)
-
-  const application = createAppObject(config)
-  const host = createHostObject()
-
-  const nameMapping = {
-    sampleRate: 'DD_TRACE_SAMPLE_RATE',
-    logInjection: 'DD_LOG_INJECTION',
-    headerTags: 'DD_TRACE_HEADER_TAGS',
-    tags: 'DD_TAGS',
-    'sampler.rules': 'DD_TRACE_SAMPLING_RULES',
-    traceEnabled: 'DD_TRACE_ENABLED',
-    url: 'DD_TRACE_AGENT_URL',
-    'sampler.rateLimit': 'DD_TRACE_RATE_LIMIT',
-    queryStringObfuscation: 'DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP',
-    version: 'DD_VERSION',
-    env: 'DD_ENV',
-    service: 'DD_SERVICE',
-    clientIpHeader: 'DD_TRACE_CLIENT_IP_HEADER',
-    'grpc.client.error.statuses': 'DD_GRPC_CLIENT_ERROR_STATUSES',
-    'grpc.server.error.statuses': 'DD_GRPC_SERVER_ERROR_STATUSES'
-  }
+module.exports = {
```

Comment:
you're mixing arrow and non arrow function ?

Reply:
i don't see any `this` so why not use only non-arrow functions ?

Reply:
Fixed, and I also did the same fix for #5254